### PR TITLE
A0-2016: Bring back dynamic fee test

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

We restore E2E test ensuring that our dynamic fees are adjusted correctly to the traffic.

In order to have higher traffic we used to use `fill_block` extrinsic. However, it was moved to a special `root_testing` pallet. While we could bring it to our runtime and use it in this test, we chose to manipulate block occupancy with a batch calls. This is because configuring this additional pallet would be a bit onerous:
 - additional feature to the node crate and the binary crate
 - two more `construct_runtime` long enum definitions (with and without `liminal` feature)
 - inconsistency with `aleph-client` (we wouldn't regenerate `aleph_zero.rs` module) in e2e tests

## Type of change

- New feature (non-breaking change which adds functionality)

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have created new documentation

### CODE REVIEW

File: e2e-tests/src/test/fee.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    High,
+}
+
+/// Ensures that the fee multiplier is adjusted according to the block occupancy.
+#[tokio::test]
+pub async fn fee_calculation() -> anyhow::Result<()> {
+    let config = setup_test();
+    let (connection, to) = setup_for_transfer(config).await;
+
+    let minimal_multiplier: FixedU128 = FixedU128::from(1);
```
Comment (kostekIV) [inline e2e-tests/src/test/fee.rs:33]: type annotation not needed?

**Review by kostekIV — APPROVED** [review]
lgtm

File: e2e-tests/src/test/fee.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+use once_cell::sync::Lazy;
+
+use crate::{config::setup_test, fee::current_fees, transfer::setup_for_transfer};
+
+/// In order to increase the block occupancy we need to transfer funds to a lot of accounts. This
+/// array contains the accounts we will be transferring funds to.
+static DESTINATIONS: Lazy<Vec<AccountId>> = Lazy::new(|| {
+    (0..1400)
+        .map(|i| keypair_from_string(&format!("//{i}")).account_id().clone())
+        .collect::<Vec<_>>()
```
[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply e2e-tests/src/test/fee.rs]: nope, leftover; removed

File: e2e-tests/src/test/fee.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply e2e-tests/src/test/fee.rs]: nope, leftover; removed

**Review by Marcin-Radecki — APPROVED** [review]

File: e2e-tests/src/test/fee.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        "High traffic should lead to higher fee"
+    );
+
+    let now = connection.get_best_block().await.unwrap().unwrap();
+    connection
+        .wait_for_block(|n| n >= now + 10, BlockStatus::Finalized)
+        .await;
+
+    let (after_traffic_fee, after_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
```
Comment (deuszx) [inline e2e-tests/src/test/fee.rs:88]: I've found myself going back and forth a couple of time to check what `None` and `1` are -- it'd be a tad more readable if these two were extracted to a value (or a `const`) and reused.

**Pull request merged**